modules/platform/nuxeo-binary-metadata/src/main/java/org/nuxeo/binary/metadata/internals/MetadataRuleDescriptor.java

```diff
@@ -108,4 +114,13 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    @Override
+    public int compareTo(MetadataRuleDescriptor o) {
+        int order = getOrder().compareTo(o.getOrder());
+        if (order != 0) {
+            return order;
+        }
+        return getId().compareTo(o.getId());
```

Comment:
Could be

    return Comparator.comparing(MetadataRuleDescriptor::getOrder)
                     .thenComparing(MetadataRuleDescriptor::getId)
                     .compare(this, o);